### ISSUE TITLE
One time expense/budget tracking

### ISSUE THREAD

**Issue by isaacgr** [firefly-iii/firefly-iii#9297]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/explanation/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

I have a budget for a project. I want to set a fixed amount and subtract from that as I use it. I don't care about month-to-month costs per se.

So for example, I have 4500 budgeted for a basement reno. I have this money available, but until we actually start working on it I wont know what we've spent overall. It might take several months to complete, I just want to subtract from that and track how much is left total and what it was spent on. Ideally I don't want to have to go into the next month and update the budget to be what's left after the previous month.

This ties into a larger "projects" budget that we have, where we've saved for several home related renovations and purchases. So we have say 10000 total, 4500 is going towards this basement reno and there are other smaller projects that we've planned out and budgeted for.

Right now this is all being managed via an excel sheet, it would be nice if there was a similar concept here in firefly.

The first thought is a no auto-budget, but with this we have to re-enter the value each month.

### Solution

I guess it would be like one time expense/budget tracking. Its sort of the compliment to a piggy bank. I've saved up this money, now I'm going to spend it on the project, or item, and see how much of it I actually use.

Additionally, it could sort of expand beyond that ideally since I could group these expenses into a larger expense, i.e. "Projects" for which we have 10k is split into "basement reno", "landscaping", "furniture" and I could track how much the 'projects' budget has left total, and what each of the 'sub projects' has used.

I would personally find this particularly useful for

- home reno projects
- large purchases
- future purchase planning (I have the piggy bank setup, I've saved the money, now what do I actually spend it on and how much of it do I use)

### What are alternatives?

I sort of think maybe subtracting from a piggy bank would be the thing. But that page says 'saved so far' and 'how much left'. I'm already there, I just want to track what actually gets spent on this item/project within this group.

As mentioned above, no auto-budget was my first go-to, but having to reset the value each month is a blocker on it.

**Comment by JC5** [comment]
On the budget page, change the date range on the top of the page to a custom range. Ie. from today until 1 january 2027 (you always have to set an end date).

Then, use the (+) button for your preferred budget and set the amount. Notice how the pop up dialog confirms your date selection.

Voila :)

**Comment by isaacgr** [comment]
> On the budget page, change the date range on the top of the page to a custom range. Ie. from today until 1 january 2027 (you always have to set an end date).
> 
> Then, use the (+) button for your preferred budget and set the amount. Notice how the pop up dialog confirms your date selection.
> 
> Voila :)

I guess thats more of a workaround though right? Like its not intuitive that those steps would yield this result, but also that setting a date range at all is part of the issue (what if it does end up going on 3 years then im back in the same boat etc.). Additionally, this ends up conflating this budget with monthly budget expenses like gas, groceries etc. The goal would be an entirely separate way to track it.

**Comment by JC5** [comment]
That's true, but you're the first to ask this and I'm not sure how much priority I can give this. 

So at least in the mean time, you have a workaround.
